### PR TITLE
feat: port MeasureTheory.Constructions.Prod.Basic

### PR DESCRIPTION
**Pull request by Komyyy**
---

- [x] depends on: #4103
- [x] depends on: #4200
* [`measure_theory.constructions.prod.basic`@`fd5edc43dc4f10b85abfe544b88f82cf13c5f844`..`3b88f4005dc2e28d42f974cc1ce838f0dafb39b8`](https://leanprover-community.github.io/mathlib-port-status/file/measure_theory/constructions/prod/basic?range=fd5edc43dc4f10b85abfe544b88f82cf13c5f844..3b88f4005dc2e28d42f974cc1ce838f0dafb39b8)

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#4103
* leanprover-community/mathlib4#4165
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/MeasureTheory/Constructions/Prod/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,804 @@
...
+    set s' : Set α := { x | ν t ≤ f x }
+    have hss' : s ⊆ s' := fun x hx => measure_mono fun y hy => hST <| mk_mem_prod hx hy
+    calc
+      μ s * ν t ≤ μ s' * ν t := mul_le_mul_right' (measure_mono hss') _
+      _ = ∫⁻ _x in s', ν t ∂μ := by rw [set_lintegral_const, mul_comm]
+      _ ≤ ∫⁻ x in s', f x ∂μ := (set_lintegral_mono measurable_const hfm fun x => id)
+      _ ≤ ∫⁻ x, f x ∂μ := (lintegral_mono' restrict_le_self le_rfl)
+      _ = μ.prod ν ST := (prod_apply hSTm).symm
+      _ = μ.prod ν (s ×ˢ t) := measure_toMeasurable _
+#align measure_theory.measure.prod_prod MeasureTheory.Measure.prod_prod
```
Comment (Komyyy) [inline Mathlib/MeasureTheory/Constructions/Prod/Basic.lean:320]: With ambiguous `· ×ˢ ·`, the timeout occurs here.

**Comment by semorrison** [comment]
Is it necessary to have this stacked on top of https://github.com/leanprover-community/mathlib4/pull/4200? It would be nice if these could proceed independently.

**Comment by Komyyy** [comment]
@semorrison The new PR is #4433.

**Pull request closed without merging**